### PR TITLE
Bring project back up to speed

### PR DESCRIPTION
**Pull request by ISibboI**
Thank you for making this amazing project! I would like to use it in my own project, however it appears that it uses an outdated version of the mongodb driver that in itself depends on an outdated version of tokio which is incompatible to newer versions.

I updated the mongodb driver to the newest version, added mongodb 5.0 to the tests, and set tokio to the newest version in the dev dependencies to ensure that the crate is now compatible with tokio 1.x.

Further I formatted the code and fixed all lints, such that all integration tests pass.

There are a few breaking changes, including adding type parameters to traits because the interface of the mongodb driver changed slightly.

I would be happy if you could release this pull request, such that I can use wither in my code soon :)

### CODE REVIEW

**Comment by simoneromano96** [comment]
Hello, I'll review asap, probably by tomorrow evening.

I should be able to merge other people's PRs

**Review by simoneromano96 — APPROVED** [review]

**Comment by simoneromano96** [comment]
LGTM

I think we should merge this, then update the docs and finally write a changelog.

**Comment by ISibboI** [comment]
Thank you very much!

**Comment by thedodd** [comment]
@ISibboI thanks for the PR and @simoneromano96 thanks for the review and merge! Very happy to see this :)

File: wither/src/model.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ const MONGO_DIFF_INDEX_BLACKLIST: [&str; 3] = ["v", "ns", "key"];
 #[async_trait]
 pub trait Model
 where
-    Self: Serialize + DeserializeOwned,
+    Self: Serialize + DeserializeOwned + Unpin + Send + Sync,
```
Comment (thedodd) [inline wither/src/model.rs:37]: @ISibboI given the updates here, the other locations where you added the `Unpin + Send + Sync` trait bounds should not be required given that they are part of the `Model` trait bound. I'm tempted to say that we should prune out the redundant trait bounds anywhere we can just to help with readability and maintenance.

**Review by thedodd — COMMENTED** [review]
I know that this has already been merged, but @ISibboI I'm wondering what your thoughts are on my comment about the redundant trait bounds. Cheers!

File: wither/src/model.rs (hunk 1 above)
Comment (ISibboI) [reply wither/src/model.rs:37]: Hi, sure. I can try to remove these bounds at all other locations.

**Comment by ndelvalle** [comment]
@thedodd @simoneromano96 can we make a release with these changes please <3

**Comment by simoneromano96** [comment]
> @thedodd @simoneromano96 can we make a release with these changes please <3

Hi, so currently I'm not able to push on master, locally I've updated the changelog to prepare the release however I cannot tag to create a new release

**Pull request merged**